src/main/webapp/app/shared/notification/notification-popup/notification-popup.component.ts

```diff
@@ -96,6 +116,24 @@ export class NotificationPopupComponent implements OnInit {
         }
     }
 
+    /**
+     * Adds a notification about a updated exercise during a live exam to the component's state
+     * and pushes updated problemStatement to student exam exercise via BehaviorSubjects
+     *
+     * @param notification {Notification}
+     */
+    private addExamUpdateNotification(notification: Notification): void {
+        if (!notification.target) return;
+
+        const target = JSON.parse(notification.target);
```

Comment:
What if this fails? Can you put it in a try catch block?

Reply:
Done. I hope this is correct.